### PR TITLE
feat: add side nav and 404 page

### PR DESCRIPTION
**Pull request by limsohee1002**
### Description
- add side nav
- add side nav api
- add local storage to save side nav data
- add 404 page

### Type(s) of changes

- [ ] Bug fix
- [x] New feature
- [ ] Update to an existing feature

### Motivation for PR
#26 #16 

### Applicable screenshots
https://www.loom.com/share/621e215a90fb4e9a9bce5bd50f060350?sid=958c8a51-a045-4cff-a1a5-d890e3f37f6c

### CODE REVIEW

File: src/modules/side-nav/SideNavModule.tsx [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+              />
+              <div key={`section-nav-${category.id}`}>
+                <h3
+                  className={cn('subheading-2 my-3', {
+                    'text-light-accent-1 dark:text-dark-accent-1': categoryisActive,
+                    'text-light-neutral-2 dark:text-dark-neutral-2': !categoryisActive,
+                  })}
+                >
+                  {category.name}
+                </h3>
```
Comment (nahbee10) [inline src/modules/side-nav/SideNavModule.tsx:111]: I think these category headings should also be clickable:
https://www.loom.com/share/96544e49e4064d78a86898d77d7b18e6?sid=2fe406d9-4a1d-40f6-9cdb-ac5657f6d2af

**Review by nahbee10 — CHANGES_REQUESTED** [review]
This overlap issue still happening! Will document this on QA ticket, let's address this later in the QA phase. 
I've added one comment about make the inactive category items are also clickable

**Comment by nahbee10** [comment]
Also noticed that we need to adjust heading level which is used by right side nav('On this page') for jump links. For now it's using too much of body text I think: 
Screenshot 2024-11-06 at 3 21 34 PM

File: src/modules/side-nav/SideNavModule.tsx (hunk 1 above)
Comment (limsohee1002) [reply src/modules/side-nav/SideNavModule.tsx]: I updated it! on nav, I think we should go though hover state with designer. I felt like it is not consistant but I guess we can handle that on QA!

**Comment by limsohee1002** [comment]
> Also noticed that we need to adjust heading level which is used by right side nav('On this page') for jump links. For now it's using too much of body text I think: Screenshot 2024-11-06 at 3 21 34 PM

right now, I am grabbing header 1 to 4 to add to side navigation. I think we should ask editor how they imagine to use that. If they would like us to only grab 1 to 3, we can do that as well, but I have seen some h4 that have different text sizes that are represented as titles, so let's confirm with the editor!
It is a super easy fix, if they want us to use h4 as side nav title, they should update this article so I think it will be good to confirm first

**Review by nahbee10 — APPROVED** [review]
We also need to add side nav items to the mobile nav modal([design](https://www.figma.com/design/QkBERlCkBKBAWRqELCC8uq/%5BUniswap%5D-%F0%9F%8D%84-Spore-Library-w%2F-Web?m=auto&node-id=5046-19167)). Let's handle it in the separate PR.
Screenshot 2024-11-06 at 9 44 45 PM

**Review by mokaymokay — APPROVED** [review]
Nice work! Just a couple small things. No additional comments besides Nahee's

File: src/modules/side-nav/SideNavModule.tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+          const categoryisActive = activeNavState.category === category.id;
+
+          return (
+            <>
+              <div
+                className={cn(
+                  'border-t border-light-neutral-2 dark:border-dark-neutral-2 opacity-10',
+                  {
+                    hidden: i === 0,
+                  }
```
[the remark this answers is not in the document]
Comment (limsohee1002) [reply src/modules/side-nav/SideNavModule.tsx:100]: It is not actually first element inside of the div 🥲 it has button div above it so I just added that ☺️

**Pull request merged**